nlptest/transform/robustness.py

```diff
@@ -610,47 +610,42 @@ def transform(sample_list: List[Sample]) -> List[Sample]:
         Returns:
             List of sentences that have numbers in their verbal representation.
         """
-
-        def convert_numbers(regex, text):
-            results = []
-            trans = []
-            transformations = []
-            start_offset = 0
-
-            for match in re.finditer(regex, text):
-                token = match.group()
-                words = NumberToWord.infEng.number_to_words(
-                    token, wantlist=True)
-                token_len = len(token) - 1
-                new_words_len = len(' '.join(words)) - 1
-                trans.append(text[start_offset:match.start()])
-                trans.append(' '.join(words))
-                start_offset = match.end()  
-                transformations.append(
-                        Transformation(
-                            original_span=Span(
-                                start=match.start(), end=match.end()-1, word=token),
-                            new_span=Span(start=match.start(), end=match.start(
-                            )+new_words_len, word=' '.join(words)),
-                            ignore=False
+        
+        def convert_numbers(regex,text):
+                results = []
+                trans = []
+                transformations = []
+                start_offset = 0
+                
+                for match in re.finditer(regex, text):
+                    token = match.group()
+                    words = NumberToWord.num.number_to_words(token, wantlist=True)
+                    new_words_len = len(' '.join(words))
+                    trans.append(text[start_offset:match.start()])
+                    trans.append(' '.join(words))
+                    start_offset = match.end()
+                    if sample.task in ("ner", "text-classification"):
+                        transformations.append(
+                            Transformation(
+                                original_span=Span(start=match.start(), end=match.end(), word=token),
+                                new_span=Span(start=match.start(), end=match.start()+new_words_len, word=' '.join(words)),
+                                ignore=False
+                            )
                         )
-                    )
-
-            trans.append(text[start_offset:])
-            results.append(''.join(trans))
-
-            return ''.join(results), trans
-
-        for idx, sample in enumerate(sample_list):
```

Comment:
don't refactor this section, this will handle when list of samples or strings in the function.